packages/arb-token-bridge-ui/src/components/TransferPanel/TokenLogo.tsx

```diff
@@ -0,0 +1,67 @@
+import { useMemo } from 'react'
+
+import { useAppState } from '../../state'
+import { useTokensFromLists, useTokensFromUser } from './TokenSearchUtils'
+import { useNetworks } from '../../hooks/useNetworks'
+import { useNetworksRelationship } from '../../hooks/useNetworksRelationship'
+import { useNativeCurrency } from '../../hooks/useNativeCurrency'
+import { SafeImage } from '../common/SafeImage'
+import { twMerge } from 'tailwind-merge'
+
+/**
+ * Shows the selected token logo by default.
+ * @param {Object} props
+ * @param {string | null} [props.srcOverride] - Optional URL to override default token logo source
+ * @param {string | undefined} [props.className] - Class name override
+ */
+export const TokenLogo = ({
+  srcOverride,
+  className
+}: {
+  srcOverride?: string | null
+  className?: string
+}) => {
+  const {
+    app: { selectedToken }
+  } = useAppState()
+  const tokensFromLists = useTokensFromLists()
+  const tokensFromUser = useTokensFromUser()
+
+  const [networks] = useNetworks()
+  const { childChainProvider } = useNetworksRelationship(networks)
+  const nativeCurrency = useNativeCurrency({ provider: childChainProvider })
+
+  const src = useMemo(() => {
+    // Override to show the native currency logo
+    if (srcOverride === null) {
+      return nativeCurrency.logoUrl
+    }
+
+    if (typeof srcOverride !== 'undefined') {
+      return srcOverride
+    }
+
+    if (selectedToken) {
+      return (
+        tokensFromLists[selectedToken.address]?.logoURI ??
+        tokensFromUser[selectedToken.address]?.logoURI
+      )
+    }
+
+    return nativeCurrency.logoUrl
+  }, [
+    nativeCurrency.logoUrl,
+    selectedToken,
+    srcOverride,
+    tokensFromLists,
+    tokensFromUser
+  ])
```

Comment:
this logic is same as `tokenLogoSrc` in `TokenButton`

maybe we can use this component in `TokenButton` directly?